tianshou/trainer/offline.py

```diff
@@ -0,0 +1,97 @@
+import time
+import tqdm
+from collections import defaultdict
+from torch.utils.tensorboard import SummaryWriter
+from typing import Dict, List, Union, Callable, Optional
+
+from tianshou.policy import BasePolicy
+from tianshou.utils import tqdm_config, MovAvg
+from tianshou.data import Collector, ReplayBuffer
+from tianshou.trainer import test_episode, gather_info
+
+
+def offline_trainer(
+    policy: BasePolicy,
+    buffer: ReplayBuffer,
+    test_collector: Collector,
```

Comment:
@Trinkle23897 I just noticed the test_collector, which needs to be initialized by an env, here is not optional. But actually, in practice, the main reason to use these offline algorithms is the lack of an env. So it might be better to make it optional. But I'm not sure what the alternative way to do the test given we don't have an env.

Reply:
Hmm... remove the test_collector and set it to an optional argument mean we don't have any evaluation metric to measure the performance of the current policy. If users don't have any runnable envs, he/she can give fake envs to test_collector.

Reply:
All right. But I feel like that is sort of hacky. Anyway, let's leave it as it is here.